pkg/actions/podidentityassociation/updater.go

```diff
@@ -106,13 +110,24 @@ func (u *Updater) update(ctx context.Context, updateConfig *UpdateConfig, podIde
 }
 
 func (u *Updater) updatePodIdentityAssociation(ctx context.Context, roleARN string, updateConfig *UpdateConfig, podIdentityAssociationID string) error {
-	if _, err := u.APIUpdater.UpdatePodIdentityAssociation(ctx, &eks.UpdatePodIdentityAssociationInput{
+	input := &eks.UpdatePodIdentityAssociationInput{
 		AssociationId: aws.String(updateConfig.AssociationID),
 		ClusterName:   aws.String(u.ClusterName),
 		RoleArn:       aws.String(roleARN),
-	}); err != nil {
-		return fmt.Errorf("(associationID: %s, roleARN: %s): %w", updateConfig.AssociationID, roleARN, err)
 	}
+
+	if updateConfig.PodIdentityAssociation.TargetRoleARN != nil {
+		input.TargetRoleArn = updateConfig.PodIdentityAssociation.TargetRoleARN
+	}
+	if updateConfig.PodIdentityAssociation.DisableSessionTags != nil {
+		input.DisableSessionTags = updateConfig.PodIdentityAssociation.DisableSessionTags
+	}
```

Comment:
do we need to do explicit nil checks if these are already the correct pointer types? Could we just assign them directly above with the rest of the input?